Firebase/DynamicLinks/FDLURLComponents/FDLURLComponents.m

```diff
@@ -448,15 +449,51 @@ - (instancetype)init {
 
 @implementation FIRDynamicLinkComponents
 
+#pragma mark Deprecated Initializers.
 + (instancetype)componentsWithLink:(NSURL *)link domain:(NSString *)domain {
   return [[self alloc] initWithLink:link domain:domain];
 }
 
 - (instancetype)initWithLink:(NSURL *)link domain:(NSString *)domain {
+  NSURL *domainURL = [NSURL URLWithString:domain];
+  if (!domainURL.scheme) {
```

Comment:
This won't log an error if someone uses an HTTP scheme, correct?

Reply:
Yes. Was looking to not add any new validation to the existing methods. Do you still want to?

Reply:
Done.

Reply:
It seems like this isn't validation but changes the functionality from how it was before. Previously we always added `https://` no matter what, but now we only add it if there's no scheme.